test cases/common/177 initial c_args/meson.build

```diff
@@ -3,5 +3,5 @@ project('options', 'c')
 # Test passing c_args and c_link_args options from the command line.
 assert(get_option('c_args') == ['-funroll-loops'],
        'Incorrect value for c_args option.')
-assert(get_option('c_link_args') == ['-random_linker_option'],
+assert(get_option('c_link_args') == ['-Wall'],
```

Comment:
This is wrong. `-Wall` is a compiler argument. It should not be used when linking. Why is it there? Does it leak from `CFLAGS`?

Reply:
Or, after reading through this more thoroughly, it should not be `-Wall` because it can come in from many places. It should be somehow a unique linker argument.

Reply:
Sure makes sense. I got `-Wall` after looking through the `cl` manpage for something that looked familiar. Maybe I should just special-case with `-Wl,` and `/link`?

Reply:
Why not stick with the old one? Does something break if you do?

Reply:
Yes because now that sanity checks use user args I need to use a real arg that won't cause the sanity check to fail.

Reply:
What about `-DSOMETHING`?

Reply:
Oh duh. That will do!